### PR TITLE
[Tutorials](fix) Remove duplicate test code from UT

### PR DESCRIPTION
**Pull request by TecJesh**
### Summary
Remove duplicate code between unit tests and tutorial, keep only one sample in tutorial. 
# New contributor declaration
- [ ] I am not making a trivial change, such as fixing a typo in a comment.

- [ ] I have written a PR description following these
  [rules](https://cbea.ms/git-commit/#why-not-how).

- [ ] I have run `pre-commit run --from-ref origin/main --to-ref HEAD`.

- Select one of the following.
  - [ ] I have added tests.
    - `/test` for `lit` tests
    - `/unittest` for C++ tests
    - `/python/test` for end-to-end tests
  - [ ] This PR does not need a test because `FILL THIS IN`.

- Select one of the following.
  - [ ] I have not added any `lit` tests.
  - [ ] The `lit` tests I have added follow these [best practices](https://mlir.llvm.org/getting_started/TestingGuide/#filecheck-best-practices),
    including the "tests should be minimal" section. (Usually running Python code
    and using the instructions it generates is not minimal.)

### CODE REVIEW

File: third_party/ascend/tutorials/03-matrix-multiplication.py [hunk 1]
```diff
@@ -182,18 +183,22 @@ def grid(META):
...
-    triton_output = matmul(a, b, activation)
-    torch_output = torch_matmul(a, b, activation)
-    print(f"triton_output_with_fp16_inputs={triton_output}")
-    print(f"torch_output_with_fp16_inputs={torch_output}")
-    torch.testing.assert_close(triton_output, torch_output, atol=1e-3, rtol=1e-3)
-    print("Passed")
+
+
+def test_matrix_multiplication():
+    for activation in ["leaky_relu_custom"]:
```
Comment (hongziqi) [inline third_party/ascend/tutorials/03-matrix-multiplication.py]: There is no need to add a for loop here. Only one item is involved.

File: third_party/ascend/tutorials/03-matrix-multiplication.py [hunk 2]
```diff
@@ -182,18 +183,22 @@ def grid(META):
...
+        for shape in [(512, 512, 512), (256, 384, 128)]:
```
Comment (hongziqi) [inline third_party/ascend/tutorials/03-matrix-multiplication.py]: why test for multi shape in here?

File: third_party/ascend/tutorials/02-fused-softmax.py [hunk 3]
```diff
@@ -108,17 +112,29 @@ def softmax(x, stream):
 # Unit Test
 # ---------
 
+
+def test_fused_softmax():
+    for shape in [(1823, 781), (128, 257)]:
```
Comment (hongziqi) [inline third_party/ascend/tutorials/02-fused-softmax.py]: Tutorials are designed purely as user guides, so generalization tests are unnecessary here.

Comment (hongziqi) [reply third_party/ascend/tutorials/02-fused-softmax.py]: It is suggested to revise all test cases and remove the generalization test entirely.

Comment (TecJesh) [reply third_party/ascend/tutorials/02-fused-softmax.py]: done

File: third_party/ascend/tutorials/07-extern-functions.py [hunk 4]
```diff
@@ -54,36 +49,36 @@ def asin_kernel(
...
 
 
-def test():
-    torch.manual_seed(0)
+def test_asin_kernel_matches_torch():
     size = 98432
+    torch.manual_seed(0)
     x = torch.rand(size, device=DEV)
-    output_triton = torch.zeros(size, device=DEV)
+    output_triton = torch.empty_like(x)
```
Comment (hongziqi) [inline third_party/ascend/tutorials/07-extern-functions.py:56]: Why is `empty_like` adopted? Tensors created via `empty_like` may contain uninitialized garbage values.

Comment (TecJesh) [reply third_party/ascend/tutorials/07-extern-functions.py:56]: All in‑bounds elements will be overwritten by tl.store, using empty_like here is safe.

**Review by hongziqi — APPROVED** [review]

**Review by KanuaK — APPROVED** [review]
